lte/gateway/c/oai/tasks/mme_app/mme_app_sgsap_location_update.c

```diff
@@ -185,339 +197,163 @@ static int _build_sgs_status(
-      itti_free(ITTI_MSG_ORIGIN_ID(message_p), message_p);
-      OAILOG_FUNC_RETURN(LOG_MME_APP, RETURNerror);
-    }
+  if (!emm_ctx_p) {
+    OAILOG_ERROR(LOG_MME_APP, "Invalid emm context \n");
+    OAILOG_FUNC_RETURN(LOG_MME_APP, RETURNerror);
   }
-
-  //LAI
-  itti_nas_location_update_acc_p->laicsfb.mccdigit2 =
-    itti_sgsap_location_update_acc->laicsfb.mccdigit2;
-  itti_nas_location_update_acc_p->laicsfb.mccdigit1 =
+  // Store LAI to be sent in Attach Accept/TAU Accept
+  emm_ctx_p->csfbparams.presencemask |= LAI_CSFB;
+  emm_ctx_p->csfbparams.lai.mccdigit1 =
     itti_sgsap_location_update_acc->laicsfb.mccdigit1;
-  itti_nas_location_update_acc_p->laicsfb.mncdigit3 =
+  emm_ctx_p->csfbparams.lai.mccdigit2 =
+    itti_sgsap_location_update_acc->laicsfb.mccdigit2;
+  emm_ctx_p->csfbparams.lai.mncdigit3 =
     itti_sgsap_location_update_acc->laicsfb.mncdigit3;
-  itti_nas_location_update_acc_p->laicsfb.mccdigit3 =
-    itti_sgsap_location_update_acc->laicsfb.mccdigit3;
-  itti_nas_location_update_acc_p->laicsfb.mncdigit2 =
-    itti_sgsap_location_update_acc->laicsfb.mncdigit2;
-  itti_nas_location_update_acc_p->laicsfb.mncdigit1 =
+  emm_ctx_p->csfbparams.lai.mncdigit1 =
     itti_sgsap_location_update_acc->laicsfb.mncdigit1;
-  itti_nas_location_update_acc_p->laicsfb.lac =
-    itti_sgsap_location_update_acc->laicsfb.lac;
+  emm_ctx_p->csfbparams.lai.mncdigit2 =
+    itti_sgsap_location_update_acc->laicsfb.mncdigit2;
+  emm_ctx_p->csfbparams.lai.mccdigit3 =
+    itti_sgsap_location_update_acc->laicsfb.mccdigit3;
+  emm_ctx_p->csfbparams.lai.lac = itti_sgsap_location_update_acc->laicsfb.lac;
+
+  OAILOG_DEBUG(
+    LOG_MME_APP,
+    "MME-APP - Mobile Identity presence mask %u \n",
+    itti_sgsap_location_update_acc->presencemask);
 
-  // Mobile Identity
+  // Store Mobile Identity to be sent in Attach Accept/TAU Accept
   if (itti_sgsap_location_update_acc->presencemask & SGSAP_MOBILE_IDENTITY) {
-    itti_nas_location_update_acc_p->presencemask |= MOBILE_IDENTITY;
+    emm_ctx_p->csfbparams.presencemask |= MOBILE_IDENTITY;
     if (
       itti_sgsap_location_update_acc->mobileid.typeofidentity ==
       MOBILE_IDENTITY_IMSI) {
-      //Convert char* IMSI/TMSI to digit format to be sent in NAS message
-      imsi_mobile_identity_t *imsi_p =
-        &itti_nas_location_update_acc_p->mobileid.imsi;
+      // Convert char* IMSI/TMSI to digit format to be sent in NAS message
+      imsi_mobile_identity_t* imsi_p = &emm_ctx_p->csfbparams.mobileid.imsi;
       MOBILE_ID_CHAR_TO_MOBILE_ID_IMSI_NAS(
         itti_sgsap_location_update_acc->mobileid.u.imsi,
         imsi_p,
         itti_sgsap_location_update_acc->mobileid.length);
     } else if (
       itti_sgsap_location_update_acc->mobileid.typeofidentity ==
       MOBILE_IDENTITY_TMSI) {
-      tmsi_mobile_identity_t *tmsi_p =
-        &itti_nas_location_update_acc_p->mobileid.tmsi;
+      tmsi_mobile_identity_t received_tmsi = {0}, *received_tmsi_p = NULL;
```

Comment:
We don't need two variables here, just "received_tmsi" is enough. Replace calls to "received_tmsi_p" below with "&received_tmsi"